### ISSUE TITLE
Use SetDataSchema for replying to schema queries

### ISSUE THREAD

**Issue by sergiimk** [kamu-data/kamu-cli#371]
In several places we are accessing data slice schemas. Prior to existence of `SetDataSchema` the only way to get it was directly from a Parquet file.

This ticket is to update such places to use schema from metadata chain.

Depends on: #314

**Comment by sergiimk** [comment]
Closing in favor of #306
